lib/cli/src/c_gen/object_file_header.rs

```diff
@@ -285,11 +288,9 @@ pub fn generate_header_file(
         value: HELPER_FUNCTIONS.to_string(),
     });
 
-    let inner_c = generate_c(&c_statements);
+    c_statements.push(CStatement::LiteralConstant {
+        value: "\n#ifdef __cplusplus\n}\n#endif\n\n".to_string(),
+    });
```

Comment:
I don't remember much about the abstractions here but it might be nice to have the CPlusPlus guard be a thing that can work on nested data / trees so we don't have to remember to close it ourselves.

Don't let that block this PR though